flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaWriterStateSerializer.java

```diff
@@ -47,8 +54,15 @@ public byte[] serialize(KafkaWriterState state) throws IOException {
     public KafkaWriterState deserialize(int version, byte[] serialized) throws IOException {
         try (final ByteArrayInputStream bais = new ByteArrayInputStream(serialized);
                 final DataInputStream in = new DataInputStream(bais)) {
-            final String transactionalIdPrefx = in.readUTF();
-            return new KafkaWriterState(transactionalIdPrefx);
+            final String transactionalIdPrefix = in.readUTF();
+            final Collection<CheckpointTransaction> ongoingTransactions = new ArrayList<>();
+            if (version >= 2) {
```

Comment:
Nit: Isn't it better to assert on version 2 and fail for other versions?

Reply:
I can do it. Both patterns exist. Using >= you could potentially downgrade again if your state is forward compatible but that's rare.